core/cas-server-core-services/src/main/java/org/apereo/cas/services/AbstractResourceBasedServiceRegistryDao.java

```diff
@@ -135,14 +142,14 @@ private void initializeRegistry(final Path configDirectory,
 
                     if (!newService.equals(oldService)) {
                         update(newService);
-                        publishEvent(new CasRegisteredServicesRefreshEvent(this));
+                        createServiceRefreshEvent.andThen(publish);
                     } else {
                         LOGGER.debug("Service [{}] loaded from [{}] is identical to the existing entry. Entry may have already been saved "
                                 + "in the event processing pipeline", newService.getId(), file.getName());
                     }
                 }
             };
-            this.serviceRegistryConfigWatcher = new ServiceRegistryConfigWatcher(serviceRegistryDirectory, onCreate, onModify, onDelete);
+            this.serviceRegistryConfigWatcher = new PathWatcher(serviceRegistryDirectory, onCreate, onModify, onDelete, INTERVAL);
```

Comment:
If this actually needs to be 0 to limit monitoring intervals, how about an overloaded constructor that removes the need for the passing of that value?